File: EPI/8.2/swift/8.2.swift
public class Node {
    var next: Node?
}

public func reverse(head: Node) -> Node {
    var previous: Node?
    var current: Node? = head

    while let c = current {
        let next = c.next
        c.next = previous
        previous = current
        current = next
    }

    return previous!
}

File: EPI/Linked Lists/Reverse a singly linked list/typescript/8.2.ts
export interface Node {
    next?: Node;
}

export function reverse(head: Node): Node {
    let prev: Node | undefined;
    let curr: Node | undefined = head;

    while (curr) {
        // This required annotation seems caused by a limitation in TS type inference.
        const next: Node | undefined = curr.next;
        curr.next = prev;
        prev = curr;
        curr = next;
    }

    return prev!;
}
